subsys/bluetooth/services/ots/ots.c

```diff
@@ -174,9 +174,10 @@ ssize_t ots_obj_name_write(struct bt_conn *conn,
 		return BT_GATT_ERR(BT_GATT_OTS_WRITE_REQUEST_REJECTED);
 	}
 
-	/* Construct a temporary name for duplication detection */
+	/* Construct a valid UTF-8 temporary name for duplication detection */
 	memcpy(name, buf, len);
 	name[len] = '\0';
+	utf8_trunc(name);
```

Comment:
Hmm, actually in line 172-174 we do a check for name length, and thus the name is never truncated (which makes sense). So as long as well have that check, we don't need this truncation.

Reply:
What if the name sent over the air is an invalid UTF-8 string? Should we store the invalid name or should we truncate it to a valid one?

Reply:
Unless the OTS specification says otherwise, I suggest that we do not modify the value sent by the client. 

Regarding the invalid UTF-8 string, I don't have a clear answer, but I suppose it doesn't hurt to just store it as it? Although it's more typical for applications to rejects invalid names, so I am fine with rejecting it too.

Reply:
Ok, I think I am just going to withdraw this PR then. I was making this change as a follow up to your comment [here](https://github.com/zephyrproject-rtos/zephyr/pull/41857/files#r787658119). Seems like after further discussion we don't think this is necessary.

Reply:
Agreed